beacon-chain/state/stateV0/getters_validator.go

```diff
@@ -19,6 +19,24 @@ import (
 	"github.com/prysmaticlabs/prysm/shared/params"
 )
 
+// IndexOutOfRangeError represents an error scenario where a validator does not exist
```

Comment:
Not mentioned in the description. Ideally these changes should be its own PR

Reply:
I had to create a custom error to be able to discern between different error cases. I agree this should come in a separate PR.

Reply:
Renamed to `ValidatorIndexOutOfRangeError` to avoid merge conflicts with #9035